Automattic/jetpack
# RNA: reuse Jetpack connection status

#### Changes proposed in this Pull Request:
* `ConnectScreen` component no longer pulls connection status from the API, and only accepts it as a property
* Jetpack connection status gets overwritten with `false` after disconnect, this PR makes it pull new connection status object via API
* Jetpack Backup and Connection UI use the HOC `withConnectionStatus` to pull the connection status

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
1. Disconnect Jetpack.
2. Go to Jetpack Backup, confirm that you see the connection screen. Click "Set up Jetpack".
3. Complete the connection, it should go well.
4. Disconnect, go to "Tools -> Connection Manager".
5. Confirm that you see the connection screen, click "Set up Jetpack".
6. Complete the connection, it should go well.
